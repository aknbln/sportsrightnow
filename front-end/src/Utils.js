export function sumTests(teamData){
    let sum = 0
    teamData.forEach((member) => {
      sum += member.unit_tests
    })
  
    return sum
  }

export function inchToFeet(inches){
  let feet = Math.floor(inches / 12)
  let remainder = inches % 12

  return feet + "\' " + remainder  + "\'"
}