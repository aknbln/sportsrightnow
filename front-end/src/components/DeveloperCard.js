import React from "react";
import Card from "react-bootstrap/Card";

function DarkText({ children }) {
  return (
    <span style={{  fontSize: '17px', color: 'black', font: 'Sans-Serif'  }}>{children}</span>
  );
}

function LightText({ children }) {
  return (
    <span style={{  fontSize: '17px', color: '#404040', font: 'Sans-Serif'  }}>{children}</span>
  );
}

const DeveloperCard = (props) => {
  const {
    name,
    image,
    gitlab_username,
    role,
    bio,
    commits,
    issues,
    unit_tests,
  } = props.data;
  return (
    <Card
    style= {{backgroundColor: 'lightgray'}}>
      <Card.Img variant="top" src={image} />
      <Card.Body>
        <Card.Title><LightText>{name}</LightText></Card.Title>

        <Card.Subtitle><LightText>@{gitlab_username}</LightText></Card.Subtitle>
        <Card.Text><LightText>Role: {role}</LightText></Card.Text>
        <Card.Text><LightText>{bio}</LightText></Card.Text>
      </Card.Body>
      <Card.Footer className="text-muted" style= {{backgroundColor: 'dimgray'}}>
        <DarkText>Commits: {commits} </DarkText><br />
        <DarkText>Issues: {issues} </DarkText><br />
        <DarkText>Unit Tests: {unit_tests}</DarkText>
      </Card.Footer>
    </Card>
  );
};

export default DeveloperCard;
