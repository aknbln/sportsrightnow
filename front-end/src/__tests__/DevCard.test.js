import renderer from 'react-test-renderer'
import DeveloperCard from '../components/DeveloperCard'
import img from '../assets/images/question.png'
import { sumTests } from '../Utils'


const DummyDev = {
    name: "Devy Dev",
    gitlab_username: "dev",
    alias: "",
    email: "dev@dev.com",
    image: img,
    role: "Morale Support",
    bio: "I'm a dev!",
    commits: 0,
    issues: 0,
    unit_tests: 0,
}

it('DevCard reads data correctly', () => {
    const component = renderer.create(<DeveloperCard data={DummyDev}/>)
    expect(component.name = "Devy Dev")
    expect(component.name = "dev@dev.com")
})

it('DevCard renders correctly', () => {
    const component = renderer.create(<DeveloperCard data={DummyDev}/>)
    let json = component.toJSON()
    expect(json).toMatchSnapshot()
})

it('Unit tests summed correctly', () => {
    const dummyData = [
        {
            name: "Bob",
            unit_tests: 1
        },
        {
            name: "Alice",
            unit_tests: 2
        },
        {
            name: "Joe",
            unit_tests: -1
        }
    ]

    let sum = sumTests(dummyData)
    expect(sum = 2)
})