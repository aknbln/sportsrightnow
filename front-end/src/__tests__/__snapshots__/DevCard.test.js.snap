// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevCard renders correctly 1`] = `
<div
  className="card"
  style={
    Object {
      "backgroundColor": "lightgray",
    }
  }
>
  <img
    className="card-img-top"
    src="question.png"
    style={
      Object {
        "height": "20vw",
        "objectFit": "cover",
        "width": "100%",
      }
    }
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      <span
        style={
          Object {
            "color": "#404040",
            "font": "Sans-Serif",
            "fontSize": "17px",
          }
        }
      >
        Devy Dev
      </span>
    </div>
    <div
      className="card-subtitle h6"
    >
      <span
        style={
          Object {
            "color": "#404040",
            "font": "Sans-Serif",
            "fontSize": "17px",
          }
        }
      >
        @
        dev
      </span>
    </div>
    <p
      className="card-text"
    >
      <span
        style={
          Object {
            "color": "#404040",
            "font": "Sans-Serif",
            "fontSize": "17px",
          }
        }
      >
        Role: 
        Morale Support
      </span>
    </p>
    <p
      className="card-text"
    >
      <span
        style={
          Object {
            "color": "#404040",
            "font": "Sans-Serif",
            "fontSize": "17px",
          }
        }
      >
        I'm a dev!
      </span>
    </p>
  </div>
  <div
    className="text-muted card-footer"
    style={
      Object {
        "backgroundColor": "dimgray",
      }
    }
  >
    <span
      style={
        Object {
          "color": "black",
          "font": "Sans-Serif",
          "fontSize": "17px",
        }
      }
    >
      Commits: 
      0
       
    </span>
    <br />
    <span
      style={
        Object {
          "color": "black",
          "font": "Sans-Serif",
          "fontSize": "17px",
        }
      }
    >
      Issues: 
      0
       
    </span>
    <br />
    <span
      style={
        Object {
          "color": "black",
          "font": "Sans-Serif",
          "fontSize": "17px",
        }
      }
    >
      Unit Tests: 
      0
    </span>
  </div>
</div>
`;
